### ISSUE TITLE
test test_bake_archive__known fails when zip is not installed

### ISSUE THREAD

**Issue by blais** [beancount/beancount#264]
**[Original report](https://bitbucket.org/blais/beancount/issue/264) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

test_bake_archive__known relies on `zip` (and `tar`):

```
ERROR: test_bake_archive__known (beancount.scripts.bake_test.TestScriptArchive)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tbm/beancount/beancount/utils/test_utils.py", line 228, in new_function
    return function(self, file.name)
  File "/home/tbm/beancount/beancount/scripts/bake_test.py", line 248, in test_bake_archive__known
    self.get_args() + [filename, outfile])
  File "/home/tbm/beancount/beancount/utils/test_utils.py", line 95, in run_with_args
    return function()
  File "/home/tbm/beancount/beancount/scripts/bake.py", line 262, in main
    archive(archival_command, output_directory, opts.output, True)
  File "/home/tbm/beancount/beancount/scripts/bake.py", line 189, in archive
    stderr=subprocess.PIPE if quiet else None)
  File "/usr/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'zip': 'zip'
```

This is documentation issue.  You need zip and tar (bean bake) and gnupg (for encryption).

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Ah yes, that's a good point. We can easily remove the dependency from zip and tar by using the corresponding Python libraries.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #264: Converted zip compression in bake from using an external zip tool to using the 'zipfile' module packaged with Python.
